var/spack/repos/builtin/packages/h5bench/package.py

```diff
@@ -70,7 +70,13 @@ def setup_build_tests(self):
         exe = os.path.basename(launcher.path)
         filename = "samples/sync-write-1d-contig-contig.json"
         filter_file(f"{exe}", f"{launcher}", filename)
-        filter_file(r"-n 1", "-n 1 --timeout 240", filename)
+        if self.spec["slurm"]:
+            filter_file(r"mpirun", "srun", filename)
+            filter_file(r"--allow-run-as-root -n 2 --oversubscribe", "-n 1", filename)
+        else:
+            filter_file(
```

Comment:
You *could* move line 75 above 77 so you aren't unnecessarily calling a file I/O operation on twice for `srun`.

Reply:
I am not sure I understood, the mpirun replacement is in another key in the JSON file.